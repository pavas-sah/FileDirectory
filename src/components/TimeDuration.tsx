import React from 'react';

import TimeAgo from 'javascript-time-ago'
import en from 'javascript-time-ago/locale/en'

// Add locale-specific relative date/time formatting rules.
TimeAgo.addLocale(en)

// Create relative date/time formatter.
const timeAgo = new TimeAgo('en-US')

export interface TimeDurationProps {
    dateInString: string
}


const TimeDuration: React.SFC<TimeDurationProps> = (props) => {
    const dt = new Date(props.dateInString);
    const dtInMS = dt.valueOf();
    const currentDateInMS = Date.now().valueOf()
    const absoluteTimeInMS = currentDateInMS - dtInMS;

    return (
    <span data-testid="time">
        {absoluteTimeInMS > 0 ? timeAgo.format(Date.now() - (absoluteTimeInMS), 'time') + " ago" :
         timeAgo.format(Date.now() - (absoluteTimeInMS), 'time') + " after" }
    </span>);

};

export default TimeDuration;