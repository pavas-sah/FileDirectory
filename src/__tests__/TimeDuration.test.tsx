import React from 'react';
import TimeDuration, { TimeDurationProps } from '../components/TimeDuration';
import renderer from 'react-test-renderer';

const DurationProps: TimeDurationProps = {
  dateInString: '10-3-2013',
};

describe('Time Duration Functional component', () => {
  it('should have some time as content.', () => {
    const timeComponent = renderer.create(<TimeDuration {...DurationProps} />);
    expect(timeComponent.root.children.toString()).toBe('7 years ago');
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<TimeDuration {...DurationProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
