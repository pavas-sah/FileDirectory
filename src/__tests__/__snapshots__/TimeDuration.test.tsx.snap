// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time Duration Functional component matches snapshot 1`] = `
<span
  data-testid="time"
>
  7 years ago
</span>
`;
